### ISSUE TITLE
Email delivery is very slow - monitor campaign delivery time?

### ISSUE THREAD

**Issue by lsergei** [mautic/mautic#7402]
We face very slow campaign delivery timings while have quite fast email delivery backend (PowerMTA). I was wondering if there is a way to see/track following data:

- email campaign started sending
- email campaign finished sending

This can help to start debugging in first place. I don't seem to see this information in campaign itself or anywhere in mautic interface. Logs don't contain anything related to issues with email delivery.

Thanks in advance.

**Comment by pety-dc** [comment]
We generate emails to file spool and then run the send command which I set to be able to run parallel. 

```
php /var/www/html/app/console mautic:emails:send --force --message-limit={LIMIT} --time-limit={TIMELIMIT}
```

They don't start parallelized, only 1 process is started each minute, but they overlap each other if the message number is higher than 1 process can handle in 1 minute. These processes are however not prepeared for parallel working, they often try to grab the same spool files. One process grabs the file, the other faces an error that it could not rename it. It would probably work with better efficiency if there was some shuffling or partitioning between spool files.

I started an issue regarding this a while ago:
https://github.com/mautic/mautic/issues/6777

I also proposed to store spool files compressed (https://github.com/mautic/mautic/issues/6776) to trade cpu to disk io/space

**Comment by lsergei** [comment]
@pety-dc Thank you. May I know how are you generating messages to spool? I am thinking to use Mautic to generate messages and write some multithreaded script to push those emails to MTA, should be definitely faster then default mode.

Also, wondering if you change delivery mode from immediate to queue and vice versa - did you have any change in delivery speed?

We got another issue when campaign suddenly just wasn't send, no reason given, logs are empty from any error. That's why I am wondering if we can see log with sending progress... Tuning message delivery speed might be hard if you have no tools to measure.

**Comment by pety-dc** [comment]
@lsergei 
I just set delivery mode to `queue` from `immediate delivery` in the Configuration.
I don't know if it has any effect on delivery speed. In my case generating the letters is a few times faster than delivering it. We use custom SMTP service. Sending emails from disk spool (queue) multithreaded is definitely faster then singlethreaded.

campaign are still a mistery to me too.

**Comment by escopecz** [comment]
The speed depends on what technology is used to send the emails. SMTP is very slow as it sends emails one by one. Email transports that send via API are much faster. They can send up to 1000 emails at one request. I know that Sparkpost transport use API. I'm not sure if others. But I wouldn't consider this to be Mautic issue. It's standard SMTP speed. Sending in parallel may help, but it will also waste your resources.

Now, I'm wondering how to help you. Are you talking about segment email or campaign email?

**Comment by html5maker** [comment]
Amazon SES also has APIs and all SES accounts have some sending limits per second (for example 50 emails/second). 
Even if you use amazon ses email configuration - Mautic cannot handle 50 emails/second and is sending 0,5 emails/second... 
It is a real shame that Mautic isn't deisgned to handle large contacts lists (100000+ contacts) because in comparison with Sendy that also uses SES and also tracks clicks - Mautic is 1000 times slower... :(

**Comment by escopecz** [comment]
Yes, SES has an API. But Mautic still use SMTP to send through SES. There is a PR that implements the API but it needs tests and a little work to do:

https://github.com/mautic/mautic/pull/6977

So it's only up to the Mautic SES users to make the transport 1000 times faster.

**Comment by html5maker** [comment]
What email providers are supported in Mautic with API? From Mautic Email Configuration it is not clear which one is using SMTP and which one uses API. 
I suggest you to update the dropdown values and add (SMTP) and (API) so users can easily understand what email provider they can use if they need to send a lot of emails (right now all such users are doing research on github and mautic forums...)

**Comment by escopecz** [comment]
Good idea. As I said earlier, I know that Sparkpost uses API. The rest must be researched and the docs or the dropdown updated.

**Comment by cswainhart** [comment]
After reading this I realize I really need to get this figured out. Using Mautic, 500,000 emails+. Amazon SES. It takes 2-3 days to clear out the Mautic spool files. Our list is growing. What happens when we top 1 million? 2 million? There is an SES API option in the Mautic email setup, but I can never get it to work.

**Comment by escopecz** [comment]
For anyone interested in getting SES API to work, please help test PR #6977

**Comment by cswainhart** [comment]
I would be interested, particularly since we are having the problem. How do I install the plugin, and configure it? We do lots of emails, over 500K on a weekly basis. So we would be a good test.

Thanks
--
Christopher Swainhart
chris@swainhart.org
+1.859.866.8048

Visit http://resourcemarketing.com

**Comment by AlbanL74fr** [comment]
@cswainhart Hi, you may want to read this thread, on the community Slack
https://mautic.slack.com/archives/CD2BVAMBR/p1552445559099600?thread_ts=1552445559.099600&cid=CD2BVAMBR

**Comment by BillPackard** [comment]
I have been suffering with Mautic settings for 3 days. I wouldn't have started it if I knew it was so slow. I'm really disappointed, probably delete it all!

**Comment by lsergei** [comment]
@BillPackard Mautic is definitely not the install-and-forget solution. However, it works with some tuning and works well.

**Comment by Power-MTA** [comment]
🙌🙌🙌 latest version powermta / pmta available with lifetime license key.
+ another tools available
Skype: salimrezabd04
Telegram: salimrezabd
Telegram: salimrezabd1
